Makefile

```diff
@@ -12,183 +9,54 @@ DATE_FMT = +%Y%m%d%H%M.%S
 # 2. 'date -r': BSD date. It does not support '-d'.
-BUILD_DATE = $(shell date -u -d "@${SOURCE_DATE_EPOCH}" "${DATE_FMT}" 2>/dev/null || date -u -r "${SOURCE_DATE_EPOCH}" "${DATE_FMT}")
+export BUILD_DATE = $(shell date -u -d "@${SOURCE_DATE_EPOCH}" "${DATE_FMT}" 2>/dev/null || date -u -r "${SOURCE_DATE_EPOCH}" "${DATE_FMT}")
+
+
+clean:
+	@rm -rf dist/
+	@docker image ls "$(DOCKER_REGISTRY)/$(DOCKER_IMAGE_NAME)*" -aq | xargs docker rmi --force
+
+dist:
+	@go run github.com/goreleaser/goreleaser@v1.14.1 release --snapshot --rm-dist
 
-GO_BUILDFLAGS ?= -trimpath -ldflags="-s -w"
+.PHONY: release
+release:
+	go run github.com/goreleaser/goreleaser@v1.14.1 release --rm-dist
 
-ifndef GOARCH
-	GOARCH=amd64
-endif
+release-notes:
+	@./.ci/release-github.sh
 
-# Transform GOARCH into the architecture of the extension layer
-ifeq ($(GOARCH),amd64)
-	ARCHITECTURE=x86_64
-else
-	ARCHITECTURE=arm64
-endif
+.PHONY: test
+test:
+	@go run gotest.tools/gotestsum@v1.9.0 --format testname --junitfile $(junitfile)
 
-export AWS_FOLDER GOARCH ARCHITECTURE DOCKER_IMAGE_NAME DOCKER_REGISTRY
+.PHONY: lint-prep
+lint-prep:
+	@go mod tidy && git diff --exit-code
 
-.PHONY: all
-all: build
+.PHONY: lint
+lint:
+	@go run github.com/golangci/golangci-lint/cmd/golangci-lint@v1.48.0 version
+	@go run github.com/golangci/golangci-lint/cmd/golangci-lint@v1.48.0 run
 
 NOTICE.txt: go.mod
 	@bash ./scripts/notice.sh
 
+.PHONY: check-linceses
 check-licenses:
-	go run github.com/elastic/go-licenser@v0.4.0 -d -exclude tf -exclude testing .
-	go run github.com/elastic/go-licenser@v0.4.0 -d -exclude tf -exclude testing -ext .java .
-	go run github.com/elastic/go-licenser@v0.4.0 -d -exclude tf -exclude testing -ext .js .
+	@go run github.com/elastic/go-licenser@v0.4.0 -d -exclude tf -exclude testing .
+	@go run github.com/elastic/go-licenser@v0.4.0 -d -exclude tf -exclude testing -ext .java .
+	@go run github.com/elastic/go-licenser@v0.4.0 -d -exclude tf -exclude testing -ext .js .
 
-update-licenses:
-	go run github.com/elastic/go-licenser@v0.4.0 -exclude tf -exclude testing .
-	go run github.com/elastic/go-licenser@v0.4.0 -exclude tf -exclude testing -ext .java .
-	go run github.com/elastic/go-licenser@v0.4.0 -exclude tf -exclude testing -ext .js .
 
-lint:
-	go run github.com/golangci/golangci-lint/cmd/golangci-lint@v1.48.0 version
-	go run github.com/golangci/golangci-lint/cmd/golangci-lint@v1.48.0 run
-
-build: check-licenses NOTICE.txt
-	CGO_ENABLED=0 GOOS=linux go build ${GO_BUILDFLAGS} -o bin/extensions/apm-lambda-extension main.go
-	cp NOTICE.txt bin/NOTICE.txt
-	cp dependencies.asciidoc bin/dependencies.asciidoc
-
-build-and-publish: check-licenses validate-layer-name validate-aws-default-region
-ifndef AWS_ACCESS_KEY_ID
-	$(error AWS_ACCESS_KEY_ID is undefined)
-endif
-ifndef AWS_SECRET_ACCESS_KEY
-	$(error AWS_SECRET_ACCESS_KEY is undefined)
-endif
-	GOARCH=${GOARCH} make build
-	GOARCH=${GOARCH} make zip
-	$(MAKE) publish
-
-zip: build
```

Comment:
**issue**: I think these tasks cannot be removed, we need a zip/build target to quickly build the extension and use the terraform script in `tf` for testing.

Reply:
I'm currently working on the smoke tests aswell and there I re-added a build rule. (See: https://github.com/elastic/apm-aws-lambda/pull/370)

You can still create the zip using `make dist`. There are ways to skip unnecessary pipes like `--skip-docker`.